src/module/item/spellcasting-entry/collection.ts

```diff
@@ -294,24 +282,24 @@ class SpellCollection<TActor extends ActorPF2e, TEntry extends BaseSpellcastingE
         // Handle signature spells, we need to add signature spells to each spell level
         if (this.entry.isSpontaneous || isFlexible) {
             for (const spell of signatureSpells) {
-                for (const result of results) {
-                    if (spell.baseRank > result.level) continue;
```

Comment:
The check here is to prevent fireball from being added to a 1st / 2nd level slot as a signature spell. Does that still work?